core/src/main/java/ch/cyberduck/core/Local.java

```diff
@@ -426,25 +426,38 @@ public String toURL() {
         return String.format("file:%s", path);
     }
 
+
     public InputStream getInputStream() throws AccessDeniedException {
-        return this.getInputStream(path);
+        return getInputStream(path);
     }
 
-    protected InputStream getInputStream(final String path) throws LocalAccessDeniedException {
+    public InputStream getInputStream(final String path) throws AccessDeniedException {
         try {
-            final FileChannel channel = FileChannel.open(Paths.get(path), StandardOpenOption.READ);
-            return new SeekableByteChannelInputStream(channel);
+            return new SeekableByteChannelInputStream(getRawChannelInputStream(path));
+        }
+        catch(RuntimeException e) {
+            throw new LocalAccessDeniedException(e.getMessage(), e);
+        }
+    }
+
+    public FileChannel getRawChannelInputStream(final String path) throws LocalAccessDeniedException {
+        try {
+            return FileChannel.open(Paths.get(path), StandardOpenOption.READ);
         }
         catch(RuntimeException | IOException e) {
             throw new LocalAccessDeniedException(e.getMessage(), e);
         }
     }
 
+    public OutputStream getOutputStream(final String path, final boolean append) throws AccessDeniedException {
+        return Channels.newOutputStream(getRawChannelOutputStream(path, append));
+    }
+
     public OutputStream getOutputStream(final boolean append) throws AccessDeniedException {
-        return this.getOutputStream(path, append);
+        return Channels.newOutputStream(getRawChannelOutputStream(path, append));
     }
 
-    protected OutputStream getOutputStream(final String path, final boolean append) throws LocalAccessDeniedException {
+    protected FileChannel getRawChannelOutputStream(final String path, final boolean append) throws LocalAccessDeniedException {
```

Comment:
Can be changed to `private` scope possibly.

Reply:
fixed